Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelper.java

```diff
@@ -0,0 +1,202 @@
+        AdhocQueryRequest adhocQueryRequest = new AdhocQueryRequest();
+        SlotType1 slotType1 = new SlotType1();
+        slotType1.setName("$XDSDocumentEntryPatientId");
+
+        ValueListType valueListType = new ValueListType();
+        valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+        slotType1.setValueList(valueListType);
+        AdhocQueryType adhocQueryType = new AdhocQueryType();
+        adhocQueryType.getSlot().add(slotType1);
+
+        adhocQueryRequest.setAdhocQuery(adhocQueryType);
+        msg.setAdhocQueryRequest(adhocQueryRequest);
+        AssertionType assertionTypeTest = new AssertionType();
+        assertionTypeTest.setMessageId("12345566");
+        HomeCommunityType homeCom = new HomeCommunityType();
+        homeCom.setHomeCommunityId("urn:oid:1.1.1.1.1.1.1");
+        assertionTypeTest.setHomeCommunity(homeCom);
+        msg.setAssertion(assertionTypeTest);
+        eventA.setMessage(msg);
+
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        verifycheckInboundPolicy(result);
+
+    }
+    
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_Outbound() {
+
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+        eventA.setDirection("outbound");
+        AdhocQueryRequestMessageType msg = new AdhocQueryRequestMessageType();
+        AdhocQueryRequest adhocQueryRequest = new AdhocQueryRequest();
+        SlotType1 slotType1 = new SlotType1();
+        slotType1.setName("$XDSDocumentEntryPatientId");
+
+        ValueListType valueListType = new ValueListType();
+        valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+        slotType1.setValueList(valueListType);
+        AdhocQueryType adhocQueryType = new AdhocQueryType();
+        adhocQueryType.getSlot().add(slotType1);
+
+        adhocQueryRequest.setAdhocQuery(adhocQueryType);
+        msg.setAdhocQueryRequest(adhocQueryRequest);
+        AssertionType assertionTypeTest = new AssertionType();
+        assertionTypeTest.setMessageId("12345566");
+        HomeCommunityType homeCom = new HomeCommunityType();
+        homeCom.setHomeCommunityId("urn:oid:1.1.1.1.1.1.1");
+        assertionTypeTest.setHomeCommunity(homeCom);
+        msg.setAssertion(assertionTypeTest);
+        eventA.setMessage(msg);
+
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        verifycheckOutboundPolicy(result);
+
+    }
+    
+    private void verifycheckOutboundPolicy(CheckPolicyRequestType actualCheckPolicy) {
+        // Verify result
+        assertNotNull(actualCheckPolicy);
+        // AssertAssertion to make sure nothing change
+        AssertionType assertTypeResult = actualCheckPolicy.getAssertion();
+
+        assertEquals("messageId should not change", "12345566", assertTypeResult.getMessageId());
+        // Assert Resource Session
```

Comment:
Declare Constants and then use it in your test class. For instance, declare MESSAGE_ID = 12345566 and then use  MESSAGE_ID where needed .